api/python/quilt3/packages.py

```diff
@@ -446,20 +446,21 @@ def browse(cls, name=None, registry=None, top_hash=None):
         # TODO: verify that name is correct with respect to this top_hash
         # TODO: allow partial hashes (e.g. first six alphanumeric)
         pkg_manifest_uri = f'{registry}/.quilt/packages/{quote(top_hash)}'
-        return cls._from_path(pkg_manifest_uri)
 
-    @classmethod
-    def _from_path(cls, uri):
-        """ Takes a URI and returns a package loaded from that URI """
-        src_url = urlparse(uri)
-        if src_url.scheme == 'file':
-            with open(parse_file_url(src_url)) as open_file:
-                pkg = cls.load(open_file)
-        elif src_url.scheme == 's3':
-            body = get_bytes(uri)
-            pkg = cls.load(io.BytesIO(body))
+        if file_is_local(pkg_manifest_uri):
+            local_pkg_manifest = parse_file_url(urlparse(pkg_manifest_uri))
         else:
-            raise NotImplementedError
+            local_pkg_manifest = CACHE_PATH / "manifest" / binascii.hexlify(pkg_manifest_uri.encode()).decode()
```

Comment:
Shouldn't there be a universal method, also used in the ObjectCacheClass, that goes from a string to the potential cache location? Because we do this in a few places and it's easy to get wrong (e.g. skip encode, don't decode, etc.)).